src/server.c

```diff
@@ -602,10 +597,26 @@ int htNeedsResize(dict *dict) {
 /* If the percentage of used slots in the HT reaches HASHTABLE_MIN_FILL
  * we resize the hash table to save memory */
 void tryResizeHashTables(int dbid) {
-    if (htNeedsResize(server.db[dbid].dict))
-        dictResize(server.db[dbid].dict);
-    if (htNeedsResize(server.db[dbid].expires))
-        dictResize(server.db[dbid].expires);
+    dbIterator dbit;
+    redisDb *db = &server.db[dbid];
+    int slot = 0;
+    dbIteratorInit(&dbit, db, MAIN_DICT);
+    if (db->resize_cursor != -1) {
+        dbit.next_slot = db->resize_cursor;
+    }
+    for (int i = 0; i < CRON_DICTS_PER_CALL; i++) {
+        slot = dbIteratorNextSlot(&dbit);
+        if (slot == -1) break;
+        if (!db->dict[slot]) break;
+        if (htNeedsResize(db->dict[slot]))
+            dictResize(db->dict[slot]);
+        if (!db->expires[slot]) break;
```

Comment:
I don't understand why these are breaks and not continues. If the database has no expires, we will check one dict per cron, which doesn't seem like the intended behavior.